conda/plugins/types.py

```diff
@@ -522,3 +524,33 @@ def __post_init__(self):
         except AttributeError:
             # AttributeError: alias is not a string
             raise PluginError(f"Invalid plugin aliases for {self!r}")
+
+
+@dataclass
+class CondaInstaller(CondaPlugin):
+    """
+    **EXPERIMENTAL**
+    Return type to use when defining a conda installer plugin hook.
+    For details on how this is used, see
+    :meth:`~conda.plugins.hookspec.CondaSpecs.conda_installers`.
+    :param name: name of the installer (e.g., ``pip``)
+    :param types: the names of the types of packages it can install (e.g. conda, pip).
+    
+    """
+
+    name: str
+    types: Iterable[str]
+    install: Callable[[str, list, EnvironmentConfig, dict], dict]
+    dry_run: Callable[[str, list, EnvironmentConfig, dict], dict]
+
+    def __post_init__(self):
+        # Handle name normalization
+        super().__post_init__()
+        
+        # Configure plugin settings for preferred_installers
+        for t in self.types:
```

Comment:
I'm not super confident about how well this works for overlapping plugin types. Needs more testing.